docroot/modules/humsci/hs_field_helpers/src/Plugin/Field/FieldType/DisplayModeField.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+namespace Drupal\hs_field_helpers\Plugin\Field\FieldType;
+
+use Drupal\Core\Entity\FieldableEntityInterface;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\Core\TypedData\DataDefinition;
+use Drupal\options\Plugin\Field\FieldType\ListItemBase;
+
+/**
+ * Plugin implementation of the 'display_mode_field' field type.
+ *
+ * @FieldType(
+ *   id = "display_mode_field",
+ *   label = @Translation("Display Mode Select"),
+ *   description = @Translation("Allow the user to choose which display mode to display the entity."),
+ *   default_widget = "options_select",
+ *   default_formatter = "list_default",
+ * )
+ */
+class DisplayModeField extends ListItemBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function storageSettingsForm(array &$form, FormStateInterface $form_state, $has_data) {
+    $element = [];
+    $default_settings = $allowed_values = $this->getSetting('allowed_values');
+
+    $element['allowed_values'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Display Modes'),
+      '#tree' => TRUE,
+      '#element_validate' => [[$this, 'elementValidate']],
+    ];
+
+    /** @var \Drupal\Core\Entity\EntityDisplayRepository $display_repo */
+    $display_repo = \Drupal::service('entity_display.repository');
```

Comment:
Shouldn't this be injected?
https://www.drupal.org/docs/8/api/services-and-dependency-injection/dependency-injection-for-a-form

Reply:
ooh. yeah. i had plans to fix that

Reply:
I found it easier to just use all the services and then do injection after it works

Reply:
Not yet possible. https://www.drupal.org/node/2053415